sw/device/silicon_creator/rom/rom.c

```diff
@@ -398,16 +445,23 @@ static rom_error_t rom_boot(const manifest_t *manifest, uint32_t flash_exec) {
   CFI_FUNC_COUNTER_INCREMENT(rom_counters, kCfiRomBoot, 1);
   HARDENED_RETURN_IF_ERROR(sc_keymgr_state_check(kScKeymgrStateReset));
 
+  keymgr_binding_value_t otp_measurement;
   const keymgr_binding_value_t *attestation_measurement =
       &manifest->binding_value;
-  uint32_t use_rom_ext_measurement =
-      otp_read32(OTP_CTRL_PARAM_OWNER_SW_CFG_ROM_KEYMGR_ROM_EXT_MEAS_EN_OFFSET);
-  if (launder32(use_rom_ext_measurement) == kHardenedBoolTrue) {
-    HARDENED_CHECK_EQ(use_rom_ext_measurement, kHardenedBoolTrue);
-    attestation_measurement =
-        (const keymgr_binding_value_t *)&boot_measurements.rom_ext;
+  uint32_t use_otp_measurement =
+      otp_read32(OTP_CTRL_PARAM_OWNER_SW_CFG_ROM_KEYMGR_OTP_MEAS_EN_OFFSET);
+  if (launder32(use_otp_measurement) == kHardenedBoolTrue) {
+    HARDENED_CHECK_EQ(use_otp_measurement, kHardenedBoolTrue);
+    rom_measure_otp_partitions(&otp_measurement);
+    SEC_MMIO_WRITE_INCREMENT(kOtpSecMmioDaiReadPartitionCreatorSwCfg +
```

Comment:
We should chat offline to see if it makes sense to track OTP reads separately to minimize the number of reads triggered by SEC_MMIO. There is probably a way to capture an OTP hash at manufacturing time and just compare that against our in memory OTP copy.